ArcMovie/ArcMovie/ViewModels/MainPageViewModel.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Prism.Commands;
+using Prism.Navigation;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+using ArcMovie.Helpers;
+using ArcMovie.Models;
+using Xamarin.Forms;
+
+namespace ArcMovie.ViewModels
+{
+    public class MainPageViewModel : BaseViewModel
+    {        
+        private int currentPage = 1;
+        private int totalPage = 0;
+
+        private List<Genre> genres;        
+        public ObservableMovieCollection<Movie> Movies { get; set; }
+
+        public DelegateCommand LoadUpcomingMoviesCommand { get; }
+        public DelegateCommand<Movie> ShowMovieDetailCommand { get; }
+        public DelegateCommand<Movie> UpdateMovieListCommand { get; }
+
+        private readonly INavigationService navigationService;
+        public MainPageViewModel(INavigationService navigationService)
+        {
+            Title = "Upcoming Movies";
+            this.navigationService = navigationService;
+            Movies = new ObservableMovieCollection<Movie>();
+
+            LoadUpcomingMoviesCommand = new DelegateCommand(async () => await ExecuteLoadUpcomingMoviesCommand().ConfigureAwait(false));
+            ShowMovieDetailCommand = new DelegateCommand<Movie>(async (Movie movie) => await ExecuteShowMovieDetailCommand(movie).ConfigureAwait(false));
+            UpdateMovieListCommand = new DelegateCommand<Movie>(async (Movie movie) => await ExecuteUpdateMovieListCommand(movie).ConfigureAwait(false));
+
+            LoadUpcomingMoviesCommand.Execute();
+        }        
+
+        private async Task ExecuteLoadUpcomingMoviesCommand()
+        {
+
+            IsBusy = true;
+
+            try
+            {
+                Movies.Clear();
+                currentPage = 1;
+                await LoadMoviesAsync(currentPage, Enums.MovieCategory.Upcoming).ConfigureAwait(false);
+            }
+            finally
+            {
+                IsBusy = false;
+            }
+        }
+
+        private async Task ExecuteShowMovieDetailCommand(Movie movie)
+        {            
+            var parameters = new NavigationParameters();
+            parameters.Add(nameof(movie), movie);
+            await navigationService.NavigateAsync("DetailPage", parameters).ConfigureAwait(false);
```

Comment:
Should probably replace hardcoded string with constant or reference.

Reply:
good point